src/Resolver/Location/FindParentStrategy.php

```diff
@@ -128,8 +135,40 @@ public function updateParent(ElementInterface $element, array $inputData): Eleme
             $newParent = DataObject::getByPath($this->fallbackPath);
         }
 
+        // If parent has not changed - no need to update.
+        if ($element->getParentId() && $element->getParentId() === $newParent?->getId()) {
+            return $element;
+        }
+
```

Comment:
Ah, good catch. You're right, I just disabled the type change on updates. 😕 

The problem i tried to solve is on the one hand, that 
- we don't have to check the status quo, as well as that
- the condition below can otherwise throw an exception on updates, if the element that we're trying to update might already be a child of a variant (e.g. it could have been moved via code below a variant - this is currently possible). If we'd only perform this check when the parent changes, then it would not be a problem.
https://github.com/pimcore/data-importer/blob/e0a48f47714d74251e47598e5bc987270e9bc794/src/Resolver/Location/FindParentStrategy.php#L144-L150
Of course you can argue, that this should actually never be the case as it's not intended to have children on variants other than variants. Although Pimcore doesn't keep you from doing this 


Here's an example

```
product
|- variant
|-- price-object   <- this cannot be updated anymore, as it's parent is a variant 
```

Reply:
I see. I would say lets accept this potential problem for now and see if it really becomes one.